gcp/src/test/java/org/apache/iceberg/gcp/biglake/BigLakeCatalogTest.java

```diff
@@ -0,0 +1,410 @@
+  }
+
+  @Test
+  public void testRemoveProperties_failWhenNamespacesAreInvalid() throws Exception {
+    assertFalse(
+        bigLakeCatalogUsingMockClient.removeProperties(
+            Namespace.of(new String[] {}), ImmutableSet.of()));
+    assertFalse(
+        bigLakeCatalogUsingMockClient.removeProperties(
+            Namespace.of(new String[] {"db", "tbl"}), ImmutableSet.of()));
+  }
+
+  @Test
+  public void testRemoveProperties_succeedForDatabase() throws Exception {
+    when(mockBigLakeClient.getDatabase(DatabaseName.of(GCP_PROJECT, GCP_REGION, CATALOG_ID, "db")))
+        .thenReturn(
+            Database.newBuilder()
+                .setHiveOptions(
+                    HiveDatabaseOptions.newBuilder()
+                        .putParameters("key1", "value1")
+                        .putParameters("key2", "value2"))
+                .build());
+
+    assertTrue(
+        bigLakeCatalogUsingMockClient.removeProperties(
+            Namespace.of(new String[] {"db"}), ImmutableSet.of("key1", "key3")));
+    verify(mockBigLakeClient, times(1))
+        .updateDatabaseParameters(
+            DatabaseName.of(GCP_PROJECT, GCP_REGION, CATALOG_ID, "db"),
+            ImmutableMap.of("key2", "value2"));
+  }
+
+  @Test
+  public void testLoadNamespaceMetadata_catalogAsExpected() throws Exception {
+    assertTrue(
+        bigLakeCatalogUsingMockClient
+            .loadNamespaceMetadata(Namespace.of(new String[] {}))
+            .isEmpty());
+    verify(mockBigLakeClient, times(1))
+        .getCatalog(CatalogName.of(GCP_PROJECT, GCP_REGION, CATALOG_ID));
+  }
+
+  @Test
+  public void testLoadNamespaceMetadata_databaseAsExpected() throws Exception {
+    when(mockBigLakeClient.getDatabase(DatabaseName.of(GCP_PROJECT, GCP_REGION, CATALOG_ID, "db")))
+        .thenReturn(
+            Database.newBuilder()
+                .setHiveOptions(
+                    HiveDatabaseOptions.newBuilder()
+                        .setLocationUri("my location uri")
+                        .putParameters("key1", "value1")
+                        .putParameters("key2", "value2"))
+                .build());
+
+    assertEquals(
+        ImmutableMap.of("location", "my location uri", "key1", "value1", "key2", "value2"),
+        bigLakeCatalogUsingMockClient.loadNamespaceMetadata(Namespace.of(new String[] {"db"})));
+  }
+
+  @Test
+  public void testLoadNamespaceMetadata_failWhenInvalid() throws Exception {
+    Exception exception =
+        assertThrows(
+            IllegalArgumentException.class,
+            () ->
+                bigLakeCatalogUsingMockClient.loadNamespaceMetadata(
+                    Namespace.of(new String[] {"n0", "n1"})));
+    assertEquals(
+        "BigLake catalog namespace can have zero (catalog) or one level (database), invalid"
+            + " namespace: n0.n1",
+        exception.getMessage());
+  }
+
+  @Test
+  public void testSetBigLakeCatalogInProperties_asExpected() throws Exception {
```

Comment:
I don't think it makes sense to test behavior using a method only exposed for testing. If you're checking whether catalog uses a property, then this should test the `initialize` that gets called by Spark rather than a different one.

Reply:
Removed this test. To test the `initialize` called by Spark without issuing a real RPC to BigLake service, we need to let the catalog accept a gRPC channel, but this is only used for test. It may not be a good idea to add this path to make the catalog more complex.